debian/rules

```diff
@@ -1,18 +1,49 @@
 #!/usr/bin/make -f
+#
+# Build Debian package using https://github.com/spotify/dh-virtualenv
+#
 
-# This file was automatically generated by stdeb 0.8.2 at
-# Fri, 12 Jun 2015 14:32:03 +0100
-export PYBUILD_NAME=matrix-synapse
-%:
-	dh $@ --with python2 --with systemd --buildsystem=pybuild --no-guessing-deps
-
-override_dh_auto_install:
-	python setup.py install --root=debian/matrix-synapse --install-layout=deb
+export DH_VIRTUALENV_INSTALL_ROOT=/opt/venvs
+# Use -march=x86-64 so any compiled C extensions are compiled with the most
+# generic as possible x64 instructions, so that compiling it on a new Intel chip
+# doesn't enable features not available on older ones or AMD.
+export CFLAGS=-march=x86-64
```

Comment:
Surely this will break on non-x86-64 architectures (like ARM or i386).  Unless we're only planning on ever supporting x86-64, in which case, \<shrug\>

Reply:
currently, we're only ever planning on shipping packages for x86-64, though I'm inclined to agree it's a shame that this will stop anyone else using our build scripts to build packages for other archs.

I'm not sure that I can think of a better way though?

Reply:
Other than having that set in the environment using `-e CFLAGS="-march=x86_64"` in the `docker run`, and testing that it gets passed through. That would probably work fine.

Reply:
One option would be to check the output of `dpkg-architecture -q DEB_HOST_ARCH`, and only set that if it's equal to `amd64`

Reply:
ok I've moved a bunch of this stuff out to a separate shell script, and made the CFLAGS conditional on `dpkg-architecture -q DEB_HOST_ARCH` as per @uhoreg 's suggestion. thanks!